### ISSUE TITLE
Only select fields that the model defines

### ISSUE THREAD

**Issue by paulcsmith** [luckyframework/record#90]
I will probably tackle this at some point, but if someone wants to give it a shot please comment here

The idea would be to change:
https://github.com/luckyframework/record/blob/5852d4b057b8f03544a6d2f231678834542a19c3/src/lucky_record/query_builder.cr#L78-L80

To something like:

```
SELECT user.name, user.age, user.email # etc. where the fields come from the fields defined on the model
```

**Comment by hanneskaeufler** [comment]
Had my fun with this in #106, feel free to pick it apart or scrap it if this is not what you had in mind!

**Comment by paulcsmith** [comment]
Closed by #106. Thanks @hanneskaeufler!
